app/user/views.py

```diff
@@ -42,3 +44,13 @@ class GroupViewSet(viewsets.ModelViewSet):
 
     def get_queryset(self):
         return Group.objects.all()
+
+
+class NotificationViewSet(SetUserMixin, DefaultPermissionsMixin,
```

Comment:
DefualtPermissionMixin uses model level permissions, where as we only need object level. Only an admin or the user can interact with this. This means you will need a new Mixin called AdminOrOwner that checks if the user is an admin or user owns the object. You will also need tests to cover this